### PR TITLE
[MO] - @ParallelSuites -> oauth

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

### Type of change

- Enhancement / new feature

### Description

This PR adds support for OAuth package-wide parallelism. Also, it fixes problems when OauthSuites is executed in re-run (implicitly solved by using @ParallelSUites instead of @IsolatedSuite)

#### Changes
- modified teardown script + additional logic deletion of Keycloak CR and CRDs. 
- more can be found in method `io.strimzi.systemtest.security.oauth.OauthAbstractST.deleteKeycloakCRDsIfPossible(ExtensionContext)`

Detailed description of method:

Provides protection and correct way how to delete CRDs of the Keycloak. Parallel execution could provide
unwanted behaviour, when we delete CRDs of the Keycloak in each OAuth class in @AfterAll. Such a case would cause the deletion of Keycloak pods, realms and more. So in each OAuth subclass, we firstly delete only CRs using {@link KeycloakUtils#deleteKeycloakWithoutCRDs(String)} and if the last suite is ending its execution, in that time the number of {@code OAUTH_NAME_SUITES} is equal to zero, which means that there is no other suite, which could be affected and we can delete also Keycloak CRDs using {@link KeycloakUtils#deleteKeycloakCRDs()}.

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by see-quick** [comment]
@strimzi-ci run tests test_only profile=regression testcase=OauthPlainST,OauthAuthorizationST,OauthScopeST,OauthTlsST

**Comment by see-quick** [comment]
@strimzi-ci run tests test_only profile=regression test_count_running_in_parallel=2 testcase=OauthPlainST,OauthAuthorizationST,OauthScopeST,OauthTlsST

File: systemtest/src/test/resources/junit-platform.properties [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # These are default values if @ParallelTest or @ParallelSuite is not specified
 # junit.jupiter.execution.parallel.mode.default <==> ParallelTest = concurrent
 # junit.jupiter.execution.parallel.mode.classes.default <==> ParallelSuite = concurrent
-junit.jupiter.execution.parallel.enabled=false
+junit.jupiter.execution.parallel.enabled=true
 junit.jupiter.execution.parallel.mode.default=same_thread
 junit.jupiter.execution.parallel.mode.classes.default=same_thread
 junit.jupiter.execution.parallel.config.strategy=fixed
-junit.jupiter.execution.parallel.config.fixed.parallelism=1
+junit.jupiter.execution.parallel.config.fixed.parallelism=10
```
Comment (im-konge) [inline systemtest/src/test/resources/junit-platform.properties:8]: Return these to defaults please.

**Review by im-konge — APPROVED** [review]

**Comment by see-quick** [comment]
@strimzi-ci run tests test_only profile=regression test_count_running_in_parallel=2 testcase=OauthPlainST,OauthAuthorizationST,OauthScopeST,OauthTlsST

**Comment by see-quick** [comment]
@strimzi-ci run tests test_only profile=regression test_count_running_in_parallel=2 testcase=OauthPlainST,OauthAuthorizationST,OauthScopeST,OauthTlsST

**Comment by see-quick** [comment]
@strimzi-ci run tests test_only profile=regression test_count_running_in_parallel=3 testcase=OauthPlainST,OauthAuthorizationST,OauthScopeST,OauthTlsST

**Comment by see-quick** [comment]
@strimzi-ci run tests test_only profile=regression test_count_running_in_parallel=2 testcase=OauthPlainST,OauthAuthorizationST,OauthScopeST,OauthTlsST

**Pull request closed without merging**